src/tests/java/htsjdk/samtools/sra/SRATest.java

```diff
@@ -0,0 +1,551 @@
+                                  "supercont2.13", "supercont2.14"),
+                    Arrays.asList(2291499, 1621675, 1575141, 1084805,
+                                  1814975, 1422463, 1399503, 1398693,
+                                  1186808, 1059964, 1561994, 774062,
+                                  756744, 926563)},
+        };
+    }
+
+    @Test(dataProvider = "testReferences")
+    public void testReferences(String acc, List<String> references, List<Integer> refLengths) {
+        if (!SRAAccession.isSupported()) return;
+
+        SamReader reader = SamReaderFactory.make().validationStringency(ValidationStringency.SILENT).open(
+                SamInputResource.of(new SRAAccession(acc))
+        );
+
+        final SAMRecordIterator samRecordIterator = reader.iterator();
+
+        SAMFileHeader header = reader.getFileHeader();
+        Set<String> headerRefNames = new TreeSet<String>();
+
+        for (SAMSequenceRecord ref : header.getSequenceDictionary().getSequences()) {
+            String refName = ref.getSequenceName();
+
+            int refIndex = references.indexOf(refName);
+            Assert.assertTrue(refIndex != -1, "Unexpected reference: " + refName);
+
+            Assert.assertEquals(refLengths.get(refIndex), (Integer) ref.getSequenceLength(), "Reference length is incorrect");
+
+            headerRefNames.add(refName);
+        }
+
+        Assert.assertEquals(headerRefNames.containsAll(references), true, "Header references are different from expected");
+
+        Set<String> requiredRefs = new HashSet<String>(references);
+        requiredRefs.removeAll(headerRefNames);
+
+        Assert.assertEquals(requiredRefs.size(), 0, "Following references are missing in SRA archive header: " +
+                requiredRefs.toString());
+
+        Set<String> foundRefNames = new HashSet<String>();
+
+        for (int i = 0; i < 10000; i++) {
+//        for (int i = 0; true; i++) {
+            boolean hasRecord = samRecordIterator.hasNext();
+            SAMRecord record = samRecordIterator.next();
+
+            Assert.assertEquals(hasRecord, record != null);
+
+            if (!hasRecord) {
+                break;
+            }
+
+            if (record.getReferenceIndex().equals(SAMRecord.NO_ALIGNMENT_REFERENCE_INDEX)) {
+                continue;
+            }
+
+            String refName = record.getReferenceName();
+            Assert.assertNotNull(refName);
+
+            Assert.assertEquals(references.contains(refName), true, "Unexpected reference name: " + refName);
+
+            if (!foundRefNames.contains(refName)) {
+                foundRefNames.add(refName);
+            }
+        }
+
+        requiredRefs = new HashSet<String>(references);
+        requiredRefs.removeAll(foundRefNames);
+
+//        Assert.assertEquals(requiredRefs.size(), 0, "Following references are missing in SRA archive: " + requiredRefs.toString());
```

Comment:
We only loaded 10,000 records form iterator. Loading all of the records might take couple of hours